elastic/kibana
# [ColorScheme] Update from avatar menu

In this PR I've added, **for Cloud users only** a dark mode switch inside the user drop down menu.  
The `"dark"` or `"light"` color scheme value is stored in the ES user profile (https://www.elastic.co/guide/en/elasticsearch/reference/current/user-profile.html) and defaults to **the space** color scheme.

## Snapshots

Light theme             |  Dark theme
:-------------------------:|:-------------------------:
![](https://github.com/elastic/kibana/assets/2854616/ad611a92-e1d8-493d-85e0-6367ed7eded4)  |  ![](https://github.com/elastic/kibana/assets/2854616/497a5bc7-1253-4a2b-891a-60d48325d0bb)

## Notes on implementation

I've updated the start contract of the `security` plugin with the following api:

* `update` // handler to update user profile settings. It is typed and changes need to extend the `UserProfileData` interface
* `userProfile$` // Observable with the user profile data
* `{ hook: useUpdateUserProfile }` // React hook to easily update a user profile from the UI + show a notification (with or without "Window reload" button)

## How to test

* Update your `kibana.yml` file with the following

```
xpack.cloud.id: "foox"
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.organization_url: "/account/"
xpack.cloud.billing_url: "/billing/"
xpack.cloud.profile_url: "/user/settings/"
```

* Modify this file `x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/maybe_add_cloud_links.ts`

```diff
--- a/x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/maybe_add_cloud_links.ts
+++ b/x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/maybe_add_cloud_links.ts
@@ -34,6 +34,7 @@ export function maybeAddCloudLinks({
   const userObservable = defer(() => security.authc.getCurrentUser()).pipe(
     // Check if user is a cloud user.
     map((user) => user.elastic_cloud_user),
+    map((user) => true),
     // If user is not defined due to an unexpected error, then fail *open*.
     catchError(() => of(true)),
     filter((isElasticCloudUser) => isElasticCloudUser === true),
```

* Launch Kibana
* You should see a dark theme switch in the user dropdown

Fixes https://github.com/elastic/kibana/issues/160534
Fixes (it doesn't) https://github.com/elastic/kibana/issues/160623
Fixes https://github.com/elastic/kibana/issues/159599
